### ISSUE TITLE
Location client stopped working with 2.18.0

### ISSUE THREAD

**Issue by jankoh** [aws/aws-sdk-java-v2#3517]
### Describe the bug

Starting with 2.18.0, the location stopped working triggering the exception:
```log
Received an UnknownHostException when attempting to interact with a service. See cause for the exact endpoint that is failing to resolve. If this is happening on an endpoint that previously worked, there may be a network connectivity issue or your DNS cache could be storing endpoints for too long.
```
Digging a bit into the problem, we found the cause to be reported as 
```log
software.amazon.awssdk.core.exception.SdkClientException: Unable to execute HTTP request: geo.eu-central-1.amazonaws.com
```
The main issue seems to be we have region (or AWS_REGION) set in the configuration, which we need to hav set to properly access AWS's S3. However, "geo.eu-central-1.amazonaws.com" simply does not exist.

Is there a change documented I have missed?

We simply initialise and use the client like that having set `~/.aws/credentials` and a working S3 using that settings.
```kotlin
private val locationClient: LocationClient = LocationClient.builder().build()
...
locationClient.searchPlaceIndexForText("find my town")
```

### Expected Behavior

LocationService should still work.

### Current Behavior

LocationService should stopped working.

### Reproduction Steps

Upgrade 2.17.X to 2.18.X (0,1, or 2)

### AWS Java SDK version used

2.18.X

### JDK version used

17.0.4

### Operating System and version

Linux 5.19.0

**Comment by debora-ito** [comment]
Hi @jankoh thank you for reaching out. Yes, that seems to be related to the Endpoints change released in 2.18.0.

Can you share a code sample we can use to repro the error?

**Comment by jankoh** [comment]
Hey @debora-ito,

I've set up a short [demo project](https://github.com/jankoh/location-demo) showing the issue. The error is triggered using 2.18, t works using the identical code in 2.17.X (just switching versions in build.gradle.kts).

**Comment by debora-ito** [comment]
Thank you, we are looking into the issue.

**Comment by zoewangg** [comment]
Thank you for reporting the issue. We are working on the fix.

**Comment by zoewangg** [comment]
Hi @jankoh, we have released a fix in 2.18.5. Could you try with the latest version?

**Comment by jankoh** [comment]
@zoewangg Tests startet working again in 2.18.5, so I can confirm the fix. Thank you!
